ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,816 @@
+            'privateChannel': isPrivate,                  // Adopt the private channel or not. Set as false by default.
+            'response': true,                             // Whether the server needs to return the receipt information of this subscription or not. Set as false by default.
+        };
+        let messageHash = name;
+        if (symbol !== undefined) {
+            const market = this.market (symbol);
+            const marketId = market['id'];
+            messageHash = name + ':' + marketId;
+            subscribe['topic'] = messageHash;
+        }
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
+    }
+
+    async watchTicker (symbol: string, params = {}) {
+        /**
+         * @method
+         * @name poloniexfutures#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://futures-docs.poloniex.com/#get-real-time-symbol-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniexfutures api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const name = '/contractMarket/ticker';
+        return await this.subscribe (name, false, symbol, params);
+    }
+
+    async watchTrades (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniexfutures#watchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @see https://futures-docs.poloniex.com/#full-matching-engine-data-level-3
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the poloniexfutures api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const name = '/contractMarket/execution';
+        symbol = this.symbol (symbol);
+        const trades = await this.subscribe (name, false, symbol, params);
+        if (this.newUpdates) {
+            limit = trades.getLimit (symbol, limit);
+        }
+        return this.filterBySinceLimit (trades, since, limit, 'timestamp', true);
+    }
+
+    async watchOrderBook (symbol: string, limit: Int = undefined, params = {}) {
```

Comment:
In watchOrderBook are we getting the full order book before returning it to the user?

Reply:
I don't know what you mean exactly, I don't know how to check for that